misc/dbt-materialize/dbt/adapters/materialize/impl.py

```diff
@@ -291,3 +291,21 @@ def get_column_schema_from_query(self, sql: str) -> List[PostgresColumn]:
         self.connections.execute(f"drop view {view_name}")
 
         return columns
+
+    @available
+    def generate_final_cluster_name(
+        self, cluster_name: str, force_deploy_suffix: bool = False
+    ) -> str:
+        cluster_name = self.execute_macro(
+            "materialize__generate_cluster_name",
```

Comment:
Does this need to include the `materialize__` prefix? I'm a little worried that will prevent people from overriding the macro. Does it not work to have the bare `generate_cluster_name`?